tools/Customization/DevHome.Customization/ViewModels/FileExplorerViewModel.cs

```diff
@@ -196,13 +196,20 @@ await Task.Run(async () =>
             {
                 _log.Error("Failed to validate source control extension");
                 return;
+            }
+
+            try
+            {
+                var wrapperResult = ExtraFolderPropertiesWrapper.Register(rootPath, typeof(SourceControlProvider).GUID);
+                if (!wrapperResult.Succeeded)
+                {
+                    _log.Error(wrapperResult.ExtendedError, "Failed to register folder for source control integration");
+                    return;
+                }
             }
-
-            var wrapperResult = ExtraFolderPropertiesWrapper.Register(rootPath, typeof(SourceControlProvider).GUID);
-            if (!wrapperResult.Succeeded)
-            {
-                _log.Error(wrapperResult.ExtendedError, "Failed to register folder for source control integration");
-                return;
+            catch (Exception ex)
+            {
+                _log.Error(ex, "An exception occurred while registering folder for File Explorer source control integration");
```

Comment:
Log info about the error message, like the message and stack trace.

Reply:
When you pass the exception as the first parameter to Log.Error, message and stack trace do show up in the log.

Reply:
> When you pass the exception as the first parameter to Log.Error, message and stack trace do show up in the log.

Cool! As a hardcore C++ dev, I love learning about all the nice quality-of-life stuff in C#.